src/SBarcodeGenerator.cpp

```diff
@@ -21,19 +22,47 @@ bool SBarcodeGenerator::generate(const QString &inputString)
         if (inputString.isEmpty()) {
             return false;
         } else {
+            // Change ecc level to max to generate image on QR code.
+            if (m_format == SCodes::SBarcodeFormat::QRCode && !m_imagePath.isEmpty()) {
+                if (m_eccLevel < 8) {
+                    qDebug() << "To draw image on QR Code use maximum level of ecc. Setting it to 8.";
+
+                    m_eccLevel = 8;
+                    emit eccLevelChanged(m_eccLevel);
```

Comment:
This works but a cleaner code would be to create a private `setEccLevel()` method. Then we don't forget to emit signal if we are going to change this member in other places in the future.